synapse/tests/test_lib_storm.py

```diff
@@ -337,6 +337,46 @@ def test_storm_deltag(self):
             self.none(node[1].get('#foo.bar'))
             self.none(node[1].get('#baz.faz'))
 
+    def test_storm_refs_ndef(self):
+        with self.getRamCore() as core:
+            pnode = core.formTufoByProp('ps:person', 32 * '0')
+            enode = core.formTufoByProp('inet:email', 'c00l@vertex.link')
+            pvalu = pnode[1].get('ps:person')
+            phas0 = core.formTufoByProp('ps:has', (pvalu, ('inet:email', 'c00l@vertex.link')))
+            core.formTufoByProp('ps:has', (pvalu, ('inet:fqdn', 'vertex.link')))
+
+            # It does not actually make sense to have a file reference a ps:has, this is just for testing
+            fnode = core.formTufoByProp('file:bytes:md5', 'd41d8cd98f00b204e9800998ecf8427e')
+            _, fvalu = s_tufo.ndef(fnode)
+            core.formTufoByProp('file:txtref', (fvalu, ('ps:has', phas0[1].get('ps:has'))))
+
+            outnodes = core.eval('ps:has refs(out)')  # out
+            outforms = sorted({node[1]['tufo:form'] for node in outnodes})
+            self.len(5, outnodes)
+            self.eq(outforms, ['inet:email', 'inet:fqdn', 'ps:has', 'ps:person'])
+
+            outnodes = core.eval('ps:has refs(out, limit=0)')  # out
+            limtoutforms = sorted({node[1]['tufo:form'] for node in outnodes})
+            self.len(2, outnodes)
+            self.eq(limtoutforms, ['ps:has'])
+
+            outnodes = core.eval('ps:has refs(out, limit=1)')  # out
+            limtoutforms = sorted({node[1]['tufo:form'] for node in outnodes})
+            self.len(3, outnodes)
+            [self.isin(form, ['inet:email', 'inet:fqdn', 'ps:has', 'ps:person']) for form in limtoutforms]
+
+            innodes = core.eval('ps:has refs(in)')  # in
+            informs = sorted({node[1]['tufo:form'] for node in innodes})
+            self.len(3, innodes)
+            self.eq(informs, ['file:txtref', 'ps:has'])  # Nothing refs in to these nodes
+
+            expected_bothforms = sorted(set(informs + outforms))
```

Comment:
could you use the ``sortedeq()`` helper here?

Reply:
Sure, didn't know about that one

Reply:
done